src/server.c

```diff
@@ -677,6 +677,42 @@ int clientsCronResizeQueryBuffer(client *c) {
     return 0;
 }
 
+/* The client output buffer can be adjusted to better fit the memory requierments.
+ *
+ * the logic is:
+ * in case the last observed peak size of the buffer equals the buffer size - we double the size
+ * in case the last observed peak size of the buffer equals less than half the buffer size - we shrink by half.
+ * in any case we will reset the peak value each cron run.
+ * The function always returns 0 as it never terminates the client. */
+int clientsCronResizeOutputBuffer(client *c) {
+
+    void *new_buf = NULL;
+
+    if(c->bufpos != 0)  return 0;
+
+    if (c->buf_usable_size > PROTO_REPLY_MIN_BYTES &&
+        c->buf_peak < c->buf_usable_size/2 ) {
+        new_buf = ztryrealloc(c->buf,c->buf_usable_size/2);
+        server.reply_buffer_shrinks++;
+
+    }
+    else if(c->buf_usable_size < PROTO_REPLY_CHUNK_BYTES &&
```

Comment:
styling.. if-else chain, the brace is with the else
```suggestion
    } else if(c->buf_usable_size < PROTO_REPLY_CHUNK_BYTES &&
```